debian/assets/carlos_ctl/tests/test_roles_driver.py

```
@@ -0,0 +1,442 @@
+        # the plan (who gets the fallback role) was persisted before the
+        # write, so the private list and the counts survive the crash
+        self.assertIn("program_plan", ledger)
+        self.run_roles(db, progress)
+        self.assertTrue(progress["roles"]["program"])
+        self.assertEqual(progress["roles"]["appended"]["program_provider"], 5)
+        self.assertEqual(self.private("roles-details.txt").count("999903"), 1)
+
+    def test_facility_refusal_fires_before_that_steps_writes(self):
+        db = FakeDb(facility=0)
+        progress = {"tables": {}}
+        with self.assertRaises(SystemExit):
```

Ruff (0.16.3)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)